devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -2253,91 +2370,112 @@ bool HumanStateProvider::impl::solveDynamicalInverseKinematics()
+                             << targetName;
+                    return false;
+                }
+                break;
+            }
+            case hde::KinematicTargetType::orientation: {
+                if (!dynamicalInverseKinematics.updateTargetOrientation(
+                        linkName, wearableTargetEntry.second->getCalibratedRotation())) {
+                    yError() << LogPrefix << "Failed to update orientation target for "
+                             << targetName;
+                    return false;
+                }
+                break;
+            }
+            case hde::KinematicTargetType::orientationAndVelocity: {
+                if (!dynamicalInverseKinematics.updateTargetOrientationAndVelocity(
+                        linkName,
+                        wearableTargetEntry.second->getCalibratedRotation(),
+                        wearableTargetEntry.second->getCalibratedAngularVelocity())) {
+                    yError() << LogPrefix << "Failed to update orientation and velocity target for "
+                             << targetName;
+                    return false;
+                }
+                break;
+            }
+            case hde::KinematicTargetType::gravity: {
+                if (!dynamicalInverseKinematics.updateTargetOrientation(
+                        linkName, wearableTargetEntry.second->getCalibratedRotation())) {
+                    yError() << LogPrefix << "Failed to update gravity target for " << targetName;
+                    return false;
+                }
+                break;
+            }
+            case hde::KinematicTargetType::floorContact: {
+                if (!dynamicalInverseKinematics.updateTargetPositionAndVelocity(
+                        linkName,
+                        wearableTargetEntry.second->getCalibratedPosition(),
+                        wearableTargetEntry.second->getCalibratedLinearVelocity())
+                    || !dynamicalInverseKinematics.updatePositionTargetAxis(
+                        linkName,
+                        {wearableTargetEntry.second->contactActive,
+                         wearableTargetEntry.second->contactActive,
+                         wearableTargetEntry.second->contactActive})) {
+
+                    yError() << LogPrefix << "Failed to update position and velocity target for "
+                             << targetName;
+                    return false;
+                }
+                break;
+            }
+            default: {
+                yError() << LogPrefix << "Invalid target type for " << targetName;
                 return false;
-                                                       }
-            break;
-        }
-        default: {
-            yError() << LogPrefix << "Invalid target type for " << targetName;
-            return false;}
+            }
         }
     }
 
-    if (!dynamicalInverseKinematics.solve(dt))
+    std::lock_guard<std::mutex> lock(mutexFlagIntegrator);
```

Comment:
This lock will reaming locked for a long time after `resetIntegrator` has been used. I think a possible solution is to introduce a local scope around the `lock_guard` declaration and the  `resetIntegrator`. An alternative is to use `std::atomic<bool>` instead of using a `bool` and `std::mutex`.

Reply:
@davidegorbani can you have a look at this?

Reply:
Yes, I can address this comment.

Reply:
I created a local scope around both the declaration of the `lockGuard` with commit https://github.com/robotology/human-dynamics-estimation/pull/388/commits/5eb0f62144b024c5cf8d32f3af27c98845882f39.